### ISSUE TITLE
`url_serde` asking for unrelated trait bounds

### ISSUE THREAD

**Issue by danjenson** [servo/rust-url#639]
Using the `serde` and `url_serde` crates, I get errors suggesting I need to fulfill unrelated trait bounds:

```
   Compiling ser v0.1.0 (/data/scratch/ser)
error[E0277]: the trait bound `for<'a> url_serde::Ser<'a, url::Url>: a::_::_serde::Serialize` is not satisfied
  --> src/b.rs:4:10
   |
4  | #[derive(Serialize, Deserialize)]
   |          ^^^^^^^^^ the trait `for<'a> a::_::_serde::Serialize` is not implemented for `url_serde::Ser<'a, url::Url>`
   | 
  ::: /home/danj/.cargo/registry/src/github.com-1ecc6299db9ec823/url_serde-0.2.0/src/lib.rs:77:46
   |
77 |     where S: Serializer, for<'a> Ser<'a, T>: Serialize
   |                                              --------- required by this bound in `url_serde::serialize`
   |
   = help: the following implementations were found:
             <url_serde::Ser<'a, std::option::Option<url::Url>> as a::_::_serde::Serialize>
             <url_serde::Ser<'a, url::Url> as a::_::_serde::Serialize>
             <url_serde::Ser<'a, url::host::Host<String>> as a::_::_serde::Serialize>
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `url_serde::De<url::Url>: a::_::_serde::Deserialize<'_>` is not satisfied
   --> src/b.rs:4:21
    |
4   | #[derive(Serialize, Deserialize)]
    |                     ^^^^^^^^^^^ the trait `a::_::_serde::Deserialize<'_>` is not implemented for `url_serde::De<url::Url>`
    | 
   ::: /home/danj/.cargo/registry/src/github.com-1ecc6299db9ec823/url_serde-0.2.0/src/lib.rs:158:40
    |
158 |     where D: Deserializer<'de>, De<T>: Deserialize<'de>
    |                                        ---------------- required by this bound in `url_serde::deserialize`
    |
    = help: the following implementations were found:
              <url_serde::De<std::option::Option<url::Url>> as a::_::_serde::Deserialize<'de>>
              <url_serde::De<url::Url> as a::_::_serde::Deserialize<'de>>
              <url_serde::De<url::host::Host> as a::_::_serde::Deserialize<'de>>
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ser`.

To learn more, run the command again with --verbose.
```

__Cargo.toml__
```
[package]
name = "ser"
version = "0.1.0"
authors = ["..."]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
serde = { version = "1.0.115", features = ["derive"] }
url = "2.1.1"
url_serde = "0.2.0"
```
__src/lib.rs__:
```
pub mod a;
pub mod b;
```
__src/a.rs__:
```
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct A {
    pub name: String,
}
```
__src/b.rs__:
```
use serde::{Deserialize, Serialize};
use url::Url;

#[derive(Serialize, Deserialize)]
pub struct B {
    #[serde(with = "url_serde")]
    uri: Url,
}
```

**Comment by djc** [comment]
Why do you think you need the `url_serde` crate? If you enable the `serde` feature for the `url` crate, that should provide working `Serialize` and `Deserialize` impls for `Url`.

**Comment by aWeinzierl** [comment]
> Why do you think you need the url_serde crate?

Me: Found no clues about that in the serde or url documentation (search at docs.rs, google, Strg+F)

**Comment by matthiasbeyer** [comment]
Same here. Nowhere is stated that the `url` crate has a `serde` feature!

Please update your documentation.

**Comment by scottroemeschke** [comment]
I also ran into this and looked through the features in the docs and serde is not mentioned.

I'd be happy to put in a PR matching the format existing features documentation though?

**Comment by hoijui** [comment]
I wanted to serialize a `Url` with serde. I spin up a web-search for "serde Url", and I got to `url_serde`.
After getting the error mentioned here, searching for solutions, I found out about the `serde` feature of `url`.
I then wanted to file a bug with `url_serde` to ask them to add deprecation notices in the README and the docu, mentioning `url`s `serde` feature as the replacement. the repository set for `url_serde` is this repo though, so .. not sure how to go about this.
